File: Number of Laser Beams in a Bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        n = len(bank)
        counter = Counter(bank[0])
        prev = counter.get("1", 0)
        count = 0
        for i in range(1, n):
            counter = Counter(bank[i])
            current = counter.get("1", 0)
            if current != 0:
                count += prev * current
                prev = current
        return count

File: Number of Laser Beams in a Bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int prev = 0;
        int lasers = 0;
        for (string& s: bank) {
            int count = 0;
            for (char ch: s) {
                count += (ch == '1');
            }
            if (count > 0) {
                lasers += prev*count;
                prev = count;
            }
        }
        return lasers;
    }
};
